inc/classes/admin/settings/class-settings.php

```diff
@@ -543,4 +565,69 @@ function( $url ) {
 			)
 		);
 	}
+
+	/**
+	 * Sanitize a font file path.
+	 *
+	 * @since 3.6
+	 *
+	 * @param  string $file Filepath to sanitize.
+	 * @return string|bool  Sanitized filepath. False if not a font file.
+	 */
+	private function sanitize_font( $file ) {
+		if ( ! is_string( $file ) ) {
+			return false;
+		}
+
+		$file = trim( $file );
+
+		if ( empty( $file ) ) {
+			return false;
+		}
+
+		$file = wp_parse_url( $file );
+
+		if ( ! empty( $file['host'] ) && $this->get_host() !== $file['host'] ) {
+			return false;
+		}
+
+		$file = '/' . trim( $file['path'], '/' );
+
+		$ext     = strtolower( pathinfo( $file, PATHINFO_EXTENSION ) );
+		$formats = [
+			'eot',
+			'otf',
+			'ttf',
+			'svg',
+			'woff',
+			'woff2',
+		];
+
+		if ( ! in_array( $ext, $formats, true ) ) {
+			return false;
+		}
+
+		return $file;
+	}
+
+	/**
+	 * Get site’s host domain.
+	 *
+	 * @since 3.6
+	 *
+	 * @return string|bool The host. False on failure.
+	 */
+	private function get_host() {
+		if ( isset( $this->host ) ) {
+			return $this->host;
+		}
```

Comment:
@Screenfeed What do you think of initializing the host property in the `__construct()`? Do you see any potential problems?

Why? Simplifies + micro-optimization. It'll remove the guard clause and allow the callback to use the property instead of calling this method each time.

Reply:
@hellofromtonya I did that for the same reason, micro-optimization.
Generally speaking, I try to never perform operations in the constructor: doing this in the constructor means we're doing this even if we don't need it. While doing this in the getter method is like lazy-loading the operation.
- In the constructor: the whole operation is done only once, but on every load, even if not needed.
- In the getter method: the whole operation is done only once again, only if needed, but requires one or several `isset()` when called.

Putting it in the constructor seems counter-productive to me but I may be wrong, you know better than me in this domain. What do you think?

Reply:
@Screenfeed Your assessment and reasoning is good and valid. It has to be weight against how often the method is called in its current structure. Then select the implementation which causes the less impact to performance (speed and memory). In this case, I suspect you are correct.

Reply:
OK, thank you Tonya.